lib/elocaps/ui/pages/history_page/game_card.dart

```diff
@@ -14,23 +15,36 @@ class GameCard extends HookConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final firstTeam = game.gamePlayers.where((element) => element.team == 1);
-    final secondTeam = game.gamePlayers.where((element) => element.team == 2);
-    final gameNotifier = ref.watch(gameProvider.notifier);
     final me = ref.watch(userProvider);
+    final myTeamNumber = game.gamePlayers
+        .where((element) => element.playerId == me.id)
+        .first
+        .team;
+    final firstTeam =
+        game.gamePlayers.where((element) => element.team == myTeamNumber);
+    final secondTeam =
+        game.gamePlayers.where((element) => element.team == 3 - myTeamNumber);
+    final gameNotifier = ref.watch(gameProvider.notifier);
     final historyNotifier = ref.read(playerHistoProvider.notifier);
     return Container(
-      width: double.infinity,
-      margin: const EdgeInsets.symmetric(horizontal: 30, vertical: 5),
-      decoration: const BoxDecoration(
-        gradient: LinearGradient(colors: [
-          Color.fromARGB(255, 115, 3, 3),
-          Color.fromARGB(255, 231, 84, 31),
-        ], begin: Alignment.bottomRight, end: Alignment.topLeft),
-        borderRadius: BorderRadius.all(Radius.circular(15)),
-      ),
-      child: Column(
-        children: [
+        width: double.infinity,
+        margin: const EdgeInsets.symmetric(horizontal: 30, vertical: 5),
+        decoration: BoxDecoration(
+          gradient: LinearGradient(colors: [
+            scoreToColor(
+                firstTeam.fold<int>(0,
+                    (previousValue, element) => previousValue + element.score),
+                game.isConfirmed,
+                game.isCancelled)[0],
+            scoreToColor(
+                firstTeam.fold<int>(0,
+                    (previousValue, element) => previousValue + element.score),
+                game.isConfirmed,
+                game.isCancelled)[1],
```

Comment:
You are doing this calculation twice, can you just return the result since it already is a list of Color as expected by `colors : ` (just remove the bracket as well)